src/material_header.F90

```diff
@@ -620,4 +700,215 @@ function openmc_material_set_densities(index, n, name, density) result(err) bind
+
+      ! Get atomic density and mass density of nuclide given atom percent
+      if (mat % atom_density(1) > ZERO) then
+        atom_density = mat % atom_density(i)
+        mass_density = mat % atom_density(i) * awr
+      ! Given weight percent
+      else
+        atom_density = -mat % atom_density(i) / awr
+        mass_density = -mat % atom_density(i)
+      end if
+
+      ! Calculate the "equivalent" atomic number Zeq of the material
+      Z_eq_sq = Z_eq_sq + atom_density * elm % Z**2
+      sum_density = sum_density + atom_density
+
+      ! Accumulate material DCS
+      dcs = dcs + atom_density * elm % Z**2 * elm % dcs
+
+      ! Accumulate material collision stopping power
+      stopping_power_collision = stopping_power_collision + mass_density &
+           * MASS_NEUTRON / N_AVOGADRO * elm % stopping_power_collision
+
+      ! Accumulate material radiative stopping power
+      stopping_power_radiative = stopping_power_radiative + mass_density &
+           * MASS_NEUTRON / N_AVOGADRO * elm % stopping_power_radiative
+    end do
+    Z_eq_sq = Z_eq_sq / sum_density
+
+    ! Calculate the positron DCS and radiative stopping power. These are
+    ! obtained by multiplying the electron DCS and radiative stopping powers by
+    ! a factor r, which is a numerical approximation of the ratio of the
+    ! radiative stopping powers for positrons and electrons. Source: F. Salvat,
+    ! J. M. Fernández-Varea, and J. Sempau, "PENELOPE-2011: A Code System for
+    ! Monte Carlo Simulation of Electron and Photon Transport," OECD-NEA,
+    ! Issy-les-Moulineaux, France (2011).
+    if (positron_) then
+      do i = 1, n_e
+        t = log(ONE + 1.0e6_8*ttb_e_grid(i)/(Z_eq_sq*MASS_ELECTRON))
+        r = ONE - exp(-1.2359e-1_8*t + 6.1274e-2_8*t**2 - 3.1516e-2_8*t**3 + &
+             7.7446e-3_8*t**4 - 1.0595e-3_8*t**5 + 7.0568e-5_8*t**6 - &
+             1.808e-6_8*t**7)
+        stopping_power_radiative(i) = r*stopping_power_radiative(i)
+        dcs(:,i) = r*dcs(:,i)
+      end do
+    end if
+
+    ! Total material stopping power
+    stopping_power = stopping_power_collision + stopping_power_radiative
+
+    ! Loop over photon energies
+    do i = 1, n_e - 1
+      w = ttb_e_grid(i)
+
+      ! Loop over incident particle energies
+      do j = i, n_e
+        e = ttb_e_grid(j)
+
+        ! Reduced photon energy
+        k = w / e
+
+        ! Find the lower bounding index of the reduced photon energy
+        i_k = binary_search(ttb_k_grid, n_k, k)
+
+        ! Get the interpolation bounds
+        k_l = ttb_k_grid(i_k)
+        k_r = ttb_k_grid(i_k+1)
+        x_l = dcs(i_k, j)
+        x_r = dcs(i_k+1, j)
+
+        ! Find the value of the DCS using linear interpolation in reduced
+        ! photon energy k
+        x = x_l + (k - k_l) * (x_r - x_l) / (k_r - k_l)
+
+        ! Ratio of the velocity of the charged particle to the speed of light
+        beta = sqrt(e*(e + TWO*MASS_ELECTRON)) / (e + MASS_ELECTRON)
```

Comment:
I'm not sure how beta is the ratio of the velocity of a charged particle to the speed of light

Reply:
I see now. Thanks!